providers/azure.go

```diff
@@ -209,6 +268,17 @@ func (p *AzureProvider) redeemRefreshToken(ctx context.Context, s *sessions.Sess
 	s.RefreshToken = jsonResponse.RefreshToken
 	s.CreatedAt = &now
 	s.ExpiresOn = &expires
+
+	// the presence of id_token depends on the configuration and
+	// verification of id_token requires oidc configuration
+	if s.IDToken != "" && p.Verifier != nil {
```

Comment:
You have this same blurb twice in redeem & refresh -- does it make sense to wrap all this in the logic of the helper method?

Reply:
refactored a bit